pkg/yurthub/filter/interfaces.go

```diff
@@ -59,4 +62,35 @@ type ObjectFilter interface {
 	Filter(obj runtime.Object, stopCh <-chan struct{}) runtime.Object
 }
 
+type FilterManager interface {
+	FindResponseFilter(req *http.Request) (ResponseFilter, bool)
+	FindObjectFilters(req *http.Request) ObjectFilter
+}
+
 type NodeGetter func(name string) (*v1.Node, error)
+
+type UnionObjectFilter []ObjectFilter
```

Comment:
It seems that we can use `filterChain` instead of defining a new UnionObjectFilter.

https://github.com/openyurtio/openyurt/blob/2432e743732df3c06c366c88d5fb7d1ee71ecdbc/pkg/yurthub/filter/responsefilter/filter.go#L217

by the way, maybe you can create a new folder named objectfilter and move `filterchain` into this folder.

Reply:
done

Reply:
@zyjhtangtang please remove UnionObjectFilter